embassy-stm32/src/dma/mod.rs

```diff
@@ -19,26 +19,24 @@ pub(crate) mod sealed {
         fn ch_num(&self) -> u8 {
             self.num() % 8
         }
-
-        fn regs(&self) -> pac::dma::Dma {
-            match self.dma_num() {
-                0 => pac::DMA1,
-                _ => pac::DMA2,
-            }
-        }
+        fn regs(&self) -> pac::dma::Dma;
```

Comment:
I wrote it in this way to later add an "AnyChannel" degraded type, similar to AnyPin: https://github.com/lulf/embassy/blob/41d838fa4e2d21d1d3be3fc19fe0c8625e9d57c9/embassy-stm32/src/gpio.rs#L166

If regs is implemented in the macro it's not possible to do that.

Reply:
Maybe an alternative is generating a `pac::DMA(n)` fn like the one for GPIO, so that this code doesn't have to know how many DMAs are there.

Reply:
Done